hack/init_build.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -u
+
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+. "${ROOT}/hack/utilities.sh" || { echo 'Cannot load bash utilities.'; exit 1; }
+
+VERSIONS="/tmp/.versions"
+GO_VERSION='1.7.3'
+HELM_VERSION='v2.0.0'
+GLIDE_VERSION='v0.12.3'
+
+function update-golang() {
+  # Check version of golang
+  local current_go_version="$(cat "${VERSIONS}/go" || echo "unknown")"
```

Comment:
go and glide would be somewhat easy, they do print something half decent for versions.
helm is weird in that it dumps something like:

```Client: &version.Version{SemVer:"v2.0.0", GitCommit:"51bdad42756dfaf3234f53ef3d3cb6bcd94144c2", GitTreeState:"clean"}```

which looks like badly formatted Go printf (all the type information). it can be parsed (imperfectly) I suppose.